include/k4a/k4a.hpp

```diff
@@ -870,10 +901,42 @@ class transformation
         }
     }
 
+    /** Transforms depth map and a custom image into the geometry of the color camera.
+     * Throws error on failure
+     *
+     * \sa k4a_transformation_depth_image_to_color_camera_custom
+     * Creates a new image with the output.
+     */
+    std::pair<image, image>
+    depth_image_to_color_camera_custom(const image &depth_image,
+                                       const image &custom_image,
+                                       k4a_transformation_interpolation_type_t interpolation_type,
+                                       uint32_t invalid_custom_value) const
+    {
+        image transformed_depth_image = image::create(K4A_IMAGE_FORMAT_DEPTH16,
+                                                      m_color_resolution.width,
+                                                      m_color_resolution.height,
+                                                      m_color_resolution.width *
+                                                          static_cast<int32_t>(sizeof(uint16_t)));
+        image transformed_custom_image = image::create(K4A_IMAGE_FORMAT_CUSTOM,
```

Comment:
This looks like it's actually incorrect. Check out docs for [k4a_transformation_depth_image_to_color_camera_custom](https://microsoft.github.io/Azure-Kinect-Sensor-SDK/develop/group___functions_gac00dd00e7612a86382e3d0a130f276bb.html#gac00dd00e7612a86382e3d0a130f276bb).

transformed_custom_image must be of CUSTOM8 or CUSTOM16. and the output image needs to be of the matching format (with a stride that's appropriate for the pixel width).

@rabbitdaxi added those functions and should probably review the usage here.

Reply:
How is this?
```cpp
image transformed_custom_image = image::create(custom_image.get_format(),
                                               m_color_resolution.width,
                                               m_color_resolution.height,
                                               m_color_resolution.width * (custom_image.get_stride_bytes() /
                                                                           custom_image.get_width_pixels()));
```

Reply:
I still don't understand how many channels of custom image requires in <code>k4a_transformation_depth_image_to_color_camera_custom()</code>.
Is it single channel?

Reply:
I believe it is only single channel. @christianmakela, is that correct?

Reply:
Single channel is correct. It can be either 8 or 16 bits per pixel.

Reply:
Yes, it is single channel, and because it can be 8 or 16bits per pixel, therefore, recommend to based on the custom_image to determine the byteperpixel for the transformed_custom_image

---
In reply to: [311326510](https://github.com/microsoft/Azure-Kinect-Sensor-SDK/pull/596#discussion_r311326510) [](ancestors = 311326510)

Reply:
This is good

---
In reply to: [311296317](https://github.com/microsoft/Azure-Kinect-Sensor-SDK/pull/596#discussion_r311296317) [](ancestors = 311296317)

Reply:
I got it. Thanks!